### PR TITLE
[NT-169] Add shipping type to Reward model

### PR DESCRIPTION
**Pull request by justinswart**
# 📲 What

Adds `ShippingType` to `Reward` to allow us to be more descriptive about the type of shipping options a particular reward has.

Related back-end PR: https://github.com/kickstarter/kickstarter/pull/16934

# 🤔 Why

Currently we only use the non-localized `summary` field that is returned with each reward's shipping info. By using an enumerable type we can maintain more meaningful localized strings at the client.

# 🛠 How

- Added `ShippingType` and `Location` to `Reward.Shipping`.
- Updated tests.

# 👀 See

| Limited shipping | Ships worldwide | US only (faked for snapshot 😅) |
| --- | --- | --- |
| image | image | image |

# ✅ Acceptance criteria

- [x] Rewards that ship to `anywhere` display _"Ships worldwide"_.
- [x] Rewards that ship to `multiple_locations` display _"Limited shipping"_.
- [x] Rewards that ship to `single_location` where that location is `United States` display _"United States only"_.
- [x] Rewards that have `no_shipping` do not display a pill for shipping.

# ⏰ TODO

- [x] Add strings for translations.

### CODE REVIEW

File: KsApi/models/Reward.swift [hunk 1]
```diff
@@ -25,14 +25,28 @@ public struct Reward {
 
   public struct Shipping {
     public let enabled: Bool
+    public let location: Location?
     public let preference: Preference?
```
Comment (justinswart) [inline KsApi/models/Reward.swift]: This is deprecated but I think we should only remove it when we remove `DeprecatedRewardPledgeViewController`.

File: Library/ViewModels/RewardCardViewModel.swift [hunk 2]
```diff
@@ -247,8 +247,24 @@ private func backerCountOrRemainingString(project: Project, reward: Reward) -> S
 }
 
 private func shippingSummaryString(project: Project, reward: Reward) -> String? {
-  if project.state == .live, reward.shipping.enabled, let shippingSummary = reward.shipping.summary {
-    return Strings.Ships_to_shipping_summary(shipping_summary: shippingSummary)
+  if project.state == .live, reward.shipping.enabled, let type = reward.shipping.type {
+    switch type {
+    case .anywhere:
+      return localizedString(key: "Ships_worldwide", defaultValue: "Ships worldwide")
```
Comment (justinswart) [inline Library/ViewModels/RewardCardViewModel.swift]: Will add these strings for translations.

File: KsApi/models/RewardTests.swift [hunk 3]
```diff
@@ -97,4 +97,51 @@ final class RewardTests: XCTestCase {
...
+
+    XCTAssertNotNil(reward)
+    XCTAssertEqual(reward.value?.id, 1)
+    XCTAssertEqual(reward.value?.description, "Some reward")
+    XCTAssertEqual(reward.value?.minimum, 10)
+    XCTAssertEqual(reward.value?.convertedMinimum, 12)
+    XCTAssertEqual(reward.value?.backersCount, 10)
+    XCTAssertEqual(true, reward.value?.shipping.enabled)
+    XCTAssertEqual(.singleLocation, reward.value?.shipping.type)
+    XCTAssertEqual(.init(id: 123, localizedName: "United States"), reward.value?.shipping.location)
```
Comment (dusi) [inline KsApi/models/RewardTests.swift]: I guess same question here about testing all the remaining cases (`noShipping` & `multipleLocations`)?

File: Library/ViewModels/RewardCardViewModel.swift [hunk 4]
```diff
@@ -247,8 +247,24 @@ private func backerCountOrRemainingString(project: Project, reward: Reward) -> S
...
+    case .multipleLocations:
+      return localizedString(key: "Limited_shipping", defaultValue: "Limited shipping")
+    case .noShipping: return nil
+    case .singleLocation:
+      if let name = reward.shipping.location?.localizedName {
+        return localizedString(
+          key: "location_name_only",
+          defaultValue: "%{location_name} only",
```
[the remark this answers is not in the document]
Comment (justinswart) [reply Library/ViewModels/RewardCardViewModel.swift]: Yeah the translator can move these around, I'm adding those strings for translation shortly 👍

File: Kickstarter-iOS/Views/RewardCardContainerViewTests.swift [hunk 5]
```diff
@@ -319,8 +319,7 @@ let allRewards: [(String, Reward)] = {
     |> Reward.lens.shipping .~ (
       .template
         |> Reward.Shipping.lens.enabled .~ true
-        |> Reward.Shipping.lens.preference .~ .restricted
-        |> Reward.Shipping.lens.summary .~ "Anywhere in the world"
+        |> Reward.Shipping.lens.type .~ .anywhere
```
[the remark this answers is not in the document]
Comment (justinswart) [reply Kickstarter-iOS/Views/RewardCardContainerViewTests.swift]: These are tested in the VM tests so I think it's ok to just have the `anywhere` case for snapshots.

File: KsApi/models/RewardTests.swift (hunk 3 above)
Comment (justinswart) [reply KsApi/models/RewardTests.swift]: Sure, let me add those.

**Pull request merged**